packages/generator/src/entity/namespace.ts

```diff
@@ -46,34 +46,26 @@ function properties(entity: VdmEntity): VariableStatementStructure[] {
   return entity.properties.map(prop => property(prop, entity));
 }
 
-function getFieldClassName(prop: VdmProperty): string {
-  return prop.isCollection ? 'CollectionField' : prop.fieldType;
-}
-
 function getFieldInitializer(
   prop: VdmProperty,
   entityClassName: string
 ): OptionalKind<VariableDeclarationStructure> {
-  const type = `'${
-    prop.edmType.startsWith('Edm')
-      ? prop.edmType
-      : prop.edmType.split('.').pop()
-  }'`;
+  const type = prop.isComplex
+    ? prop.jsType
+    : `'${
+        prop.edmType.startsWith('Edm')
+          ? prop.edmType
+          : prop.edmType.split('.').pop()
+      }'`;
 
-  const className = getFieldClassName(prop);
-  const thirdParameterSingle = prop.isComplex ? undefined : type;
-  const thirdParameter = prop.isCollection
-    ? createPropertyFieldInitializer(
-        prop.fieldType,
-        '',
-        entityClassName,
-        thirdParameterSingle
-      )
-    : thirdParameterSingle;
+  const className = prop.fieldType;
+  const genericParameters = getGenericParameters(entityClassName, prop);
+  const thirdParameter =
```

Comment:
What is this doing? The name `thirdParameter` is not really good. If you though it anyway perhaps you can give it a name, which relates to the meaning.

Reply:
I know it was there before #boyScout

Reply:
I did this before because there are different possibilities as to what the third parameter is. Sometimes it is one thing, sometimes another. I'll check whether with my changes I can improve that.

Reply:
done 😊